session.go

```diff
@@ -503,12 +495,11 @@ func (s *session) closeImpl(e error, remoteClose bool) error {
 	if !atomic.CompareAndSwapUint32(&s.closed, 0, 1) {
 		return errSessionAlreadyClosed
 	}
-	s.closeErr = e
 
 	if e == errCloseSessionForNewVersion {
 		s.streamsMap.CloseWithError(e)
 		s.closeStreamsWithError(e)
-		s.closeChan <- nil
+		s.closeChan <- e
```

Comment:
All paths leaving this function send `e` on the closeChan. I think it would be nicer to `defer s.closeChan <- e` at the top of the function. We then have to move the `e==nil` case to the top, but that shouldn't change anything.